[Mobile] - Refactor gallery - Cleanup

mkevins opened the pull request:
### Related PRs

* Main refactor PR: https://github.com/WordPress/gutenberg/pull/25940
* Mobile refactor PR: https://github.com/WordPress/gutenberg/pull/31306
  * This is the main mobile PR which is targeting the main refactor PR to bring the changes needed for the mobile implementations.
* Former mobile refactor PR: https://github.com/WordPress/gutenberg/pull/26823
  * Changes from this PR are being extracted and included in this and subsequent PRs

## Description
This PR takes care of some "cleanup" issues, left over from the previous PRs. It adds some props / styles that were inadvertantly removed in prior PRs, and also resumes rendering of the loading spinner for imageSizeOptions on web, which we do not display on mobile.

## How has this been tested?
The web gallery (both the old version and the new one) should render and behave exactly the same as they do in the main web gallery PR. In particular, the loading spinner should be present in the controls when the imageSizeOptions are still loading.

Mobile gallery should behave the same as the main mobile PR, and should not result in changes in rendering for other blocks that use the block list, and utilize contentResizeMode 'stretch'.

Note: This also brings in a small addition to the Platform module, [discussed here](https://github.com/WordPress/gutenberg/pull/18058#issuecomment-586118076).

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

mkevins committed:
Add back blockWidth contentContainerStyles in block list

I believe this was inadvertantly removed in some earlier commits, so
this commit adds it back.

mkevins committed:
Use boolean flags for variants in Platform module

These flags allow for a slightly more flexible, performant, and terse
way of branching by platform. For more details, see:
https://github.com/WordPress/gutenberg/pull/18058#issuecomment-586118076

mkevins committed:
Use boolean Platform flags

mkevins committed:
Only render imageSizeOptions loading spinner on web

mkevins committed:
Add default for destructured context in image edit

This is necessary for unit tests, because they instantiate the block's
edit component directly, and so the default context is not provided.

mkevins committed:
Merge branch 'rnmobile/refactor/gallery-as-nested-image-blocks' into rnmobile/refactor/gallery-as-nested-image-blocks-cleanup

antonis reviewed (approved):
Great work @mkevins 👍 
Thank you for taking care of this. I tested it running metro and everything works as expected and looks in sync with the web 🎉 
Thought in draft I'm approving this now. I can recheck if extra tweaks are needed as discussed.

mkevins committed:
Temporarily hard-code experimenal gallery refactor flag to true

This will be reverted once the block settings are fetched from the REST
API. This is enabled for now for testing purposes.

mkevins committed:
Use non-deprecated hook when possible

This brings the same change from this PR:
https://github.com/WordPress/gutenberg/pull/31027 which may have missed
the deprecated hook in the refactor PR, since it hadn't landed yet.

mkevins committed:
Merge branch 'rnmobile/refactor/gallery-as-nested-image-blocks' into rnmobile/refactor/gallery-as-nested-image-blocks-cleanup

mkevins committed:
Merge branch 'rnmobile/refactor/gallery-as-nested-image-blocks' into rnmobile/refactor/gallery-as-nested-image-blocks-cleanup

mkevins committed:
Revert "Temporarily hard-code experimenal gallery refactor flag to true"

This reverts commit 6fbaed9ec11de7b3a4bb511da561f7e8d4a76cad.

mkevins committed:
Update experiments page with warning about the mobile app version

mkevins commented:
Thank you for reviewing and testing @antonis :+1:
